test/distributed/fsdp/test_fsdp.py

```diff
@@ -0,0 +1,259 @@
+        self.register_buffer("vocab_bias", self.embed_tokens.weight.new_ones((d_model,)))
+        self.register_buffer("long_buffer", torch.zeros_like(self.vocab_bias, dtype=torch.long))  # type: ignore[arg-type]
+
+        self.bs = 2
+        self.bn = torch.nn.BatchNorm1d(self.bs) if add_bn else torch.nn.Identity()
+
+    def get_input(self, device):
+        torch.manual_seed(1 + self.rank)  # keep everything deterministic
+        src = torch.arange(12, device=device).view(6, self.bs)  # T x B
+        tgt = torch.arange(self.bs * 4, device=device).view(4, self.bs)  # T x B
+        return (src, tgt)
+
+    def forward(self, src_ids, tgt_ids):
+        src = self.embed_tokens(src_ids)
+        src = src + self.vocab_bias + self.long_buffer.type_as(src)  # type: ignore[operator]
+        tgt = self.embed_tokens(tgt_ids)
+        tgt = self.bn(tgt)
+        x = self.transformer(src, tgt)
+        return self.output_proj(x)
+
+    def get_loss(self, input, output):
+        _, tgt = input
+        return nn.functional.cross_entropy(output.view(-1, output.size(-1)), tgt.view(-1), reduction="sum")
+
+    def run_backward(self, loss):
+        loss.backward()
+
+
+class NestedWrappedModule(nn.Module):
+    def __init__(self, group, wrap_fsdp, wrap_everything=False):
+        super().__init__()
+        self.rank = group.rank()
+        self.world_size = group.size()
+
+        def _maybe_wrap(layer):
+            if wrap_fsdp:
+                return FullyShardedDataParallel(layer, group)
+            return layer
+
+        torch.manual_seed(0)  # keep everything deterministic
+
+        if wrap_everything:
+            self.module = nn.Sequential(
+                _maybe_wrap(nn.Linear(8, 4)),
+                _maybe_wrap(nn.Linear(4, 16)),
+                _maybe_wrap(nn.Linear(16, 4)),
+                _maybe_wrap(nn.Linear(4, 8)),
+            )
+        else:
+            self.module = nn.Sequential(
+                nn.Linear(8, 4),
+                _maybe_wrap(nn.Sequential(_maybe_wrap(nn.Linear(4, 16)), nn.Linear(16, 16),)),
+                _maybe_wrap(nn.Linear(16, 4)),
+                nn.Linear(4, 8),
+            )
+
+    def get_input(self, device):
+        torch.manual_seed(1 + self.rank)  # keep everything deterministic
+        return (torch.rand(4, 8, device=device),)
+
+    def forward(self, x):
+        return self.module(x)
+
+    def get_loss(self, input, output):
+        loss = output.sum()
+        return loss
+
+    def run_backward(self, loss):
+        loss.backward()
+
+
+class FSDPTest(MultiProcessTestCase):
+    def setUp(self):
+        super(FSDPTest, self).setUp()
+        self._spawn_processes()
+
+    @property
+    def world_size(self):
+        return torch.cuda.device_count() if torch.cuda.is_available() else 4
+
+    @property
+    def init_method(self):
+        return "{}{file_name}".format(FILE_SCHEMA, file_name=self.file_name)
+
+    @classmethod
+    def _run(cls, rank, test_name, file_name, pipe):
```

Comment:
Why do you need to override this internal method? I don't see other distributed tests doing it?